### PR TITLE
Synchronise settings and global "is-powered" state

### PR DESCRIPTION
**Pull request by jeremypw**
After merging this, it is necessary to merge https://github.com/elementary/switchboard-plug-bluetooth/pull/206 to allow switchboard to continue to sync properly.

After merging both PRs, all four switches affecting the Bluetooth power state *should* remain synchonised and stable.

To test:

1) Check that changing the Bluetooth enabled state by any of the switches in Wingpanel, Switchboard or `io.elementary.desktop.wingpanel.bluetooth.bluetooth-enabled` results in the other states changing accordingly.

2) Suspend and change the Bluetooth enabled state using the greeter wingpanel.  This should operate stably.  On reawakening, the changed state should be reflected in the session wingpanel.

3) Restart and check greeter Bluetooth indicator works as expected

### CODE REVIEW

File: src/Services/Manager.vala [hunk 1]
```diff
@@ -53,6 +53,14 @@ public class BluetoothIndicator.Services.ObjectManager : Object {
             object_manager.object_removed.connect ((object) => {
                 object.get_interfaces ().foreach ((iface) => on_interface_removed (object, iface));
             });
+
+            notify["is-powered"].connect (() => {
+                settings.set_boolean ("bluetooth-enabled", this.is_powered);
+            });
+
+            settings.changed["bluetooth-enabled"].connect (() => {
+                set_state_from_settings ();
```
Comment (lenemter) [inline src/Services/Manager.vala:62]:
```suggestion
                set_state_from_settings.begin ();
```

**Comment by jeremypw** [comment]
Fixing CI ...

**Comment by jeremypw** [comment]
This was found to reintroduce #192 :disappointed: so converted back to draft.

**Comment by jeremypw** [comment]
Need to revert unused change.

**Comment by zeebok** [comment]
Code looks good, I just need to test!

**Comment by jeremypw** [comment]
@zeebok Thanks for reviewing! Remember that for everything to work properly you need to install the corresponding switchboard plug PR as well :wink:

**Review by zeebok — APPROVED** [review]
Works with the other PR just as advertised. Great work!

**Comment by jeremypw** [comment]
Blocking merge until other https://github.com/elementary/switchboard-plug-bluetooth/pull/206 approved - they can then be merged at the same time.

**Comment by zeebok** [comment]
Oops that is totally my bad, I forgot to approve the other one! Should be all set now!

**Pull request merged**